system/Cache/CacheFactory.php

```diff
@@ -93,7 +94,20 @@ public static function getHandler($config, string $handler = null, string $backu
 			}
 		}
 
-		$adapter->initialize();
+		// If $adapter->initialization throws a CriticalError exception, we will attempt to
+		// use the $backup handler, if that also fails, we resort to the dummy handler.
+		try
+		{
+			$adapter->initialize();
+		}
+		catch (CriticalError $e)
+		{
+			// log the fact that an exception occurred as well what handler we resorting to
+			log_message('critical', $e->getMessage() . ' Resorting to using ' . $backup . ' handler.');
+
+			// get the next best cache handler (or dummy if the $backup also fails)
+			$adapter = self::getHandler($config, $backup, 'dummy');
```

Comment:
Doesn't the adapter need initialized here once we grab a new one?

Reply:
Line 109 returns an adapter that has already been initialized.